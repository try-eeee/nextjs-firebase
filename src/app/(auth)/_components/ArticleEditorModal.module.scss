.form {
  padding: 1rem;
}

.header {
  padding: 1em;
  background-color: #cccccc;
  font-weight: 700;
  display: flex;
  justify-content: space-between;
  align-items: center;

  .closeIcon {
    cursor: pointer;
  }
}

.footer {
  border-top: 1px solid #cccccc;
  padding: 1em;
  display: flex;
  justify-content: flex-end;
}

.viewLabel {
  padding-bottom: 0.5rem;
}

.viewTitle {
  padding-bottom: 1rem;
}

.viewContent {
  padding-bottom: 1rem;
}

.error {
  color: red;
  font-size: 0.875rem;
  display: flex;
  justify-content: flex-end;
}
