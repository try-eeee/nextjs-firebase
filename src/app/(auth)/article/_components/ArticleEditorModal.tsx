"use client";

import clsx from "clsx";
import { Button } from "@/components/button/Button";
import styles from "./ArticleEditorModal.module.scss";
import { Dialog, useDialog } from "@/components/dialog/Dialog";
import { Textarea } from "@/components/textarea/Textarea";
import { zodResolver } from "@hookform/resolvers/zod";
import { RefObject, useEffect } from "react";
import { useForm } from "react-hook-form";
import { articleEditorSchema } from "../_utils/articleEditorSchema";
import {
  createArticle,
  CreateArticle,
  deleteArticle,
  updateArticle,
} from "../actions";
import { Article } from "../_utils/asyncFunctions";
import { Input } from "@/components/input/Input";
import { Label } from "@/components/label/Label";
import { FirebaseError } from "firebase-admin";
import { IoMdClose } from "react-icons/io";

export type OperationType = "edit" | "create" | "delete" | "view";

interface ArticleEditorModalProps {
  operationType: OperationType;
  aricleRegisterModal: RefObject<HTMLDialogElement>;
  article?: Article;
  uid: string;
}

export function ArticleEditorModal(props: ArticleEditorModalProps) {
  const { operationType, aricleRegisterModal, article, uid } = props;
  const { ref: myRef, closeModal } = useDialog(aricleRegisterModal);

  useEffect(() => {
    reset({
      title: article?.title ?? "",
      content: article?.content ?? "",
    });
  }, [article]);

  const {
    register,
    handleSubmit,
    setValue,
    getValues,
    reset,
    formState: { errors },
  } = useForm<CreateArticle>({
    mode: "onSubmit",
    resolver: zodResolver(articleEditorSchema),
  });

  const onSubmit = async (data: CreateArticle) => {
    try {
      await {
        create: () => createArticle({ ...data, uid }),
        edit: () =>
          updateArticle({
            ...data,
            articleId: article?.articleId ?? "",
            uid,
          }),
        delete: () => {
          return;
        },
        view: () => {
          return;
        },
      }[operationType]();
    } catch (e) {
      const errorCode = (e as FirebaseError).code;
      const errorMessage = (e as FirebaseError).message;
      console.error(`Error Code: ${errorCode}, Message: ${errorMessage}`);
    } finally {
      reset();
      closeModal();
    }
  };

  const handleDelete = async () => {
    if (!article?.articleId) {
      console.error("");
      return;
    }
    try {
      await deleteArticle({
        uid,
        articleId: article.articleId,
      });
    } catch (e) {
      const errorCode = (e as FirebaseError).code;
      const errorMessage = (e as FirebaseError).message;
      console.error(`Error Code: ${errorCode}, Message: ${errorMessage}`);
    } finally {
      reset();
      closeModal();
    }
  };

  const handleCloseModal = () => {
    reset();
    closeModal();
  };

  return (
    <Dialog ref={myRef} handleOutside={handleCloseModal}>
      <>
        <div
          className={clsx(
            styles.header,
            {
              create: styles.create,
              edit: styles.edit,
              delete: styles.delete,
              view: styles.view,
            }[operationType]
          )}
        >
          <span>
            {
              {
                create: "新規作成",
                edit: "編集",
                delete: "削除",
                view: "詳細",
              }[operationType]
            }
          </span>
          <IoMdClose className={styles.closeIcon} onClick={handleCloseModal} />
        </div>
        <div className={styles.form}>
          {operationType === "view" ? (
            <div className={styles.viewLabel}>タイトル</div>
          ) : (
            <Label htmlFor="title" align="left" required>
              タイトル
            </Label>
          )}
          {operationType === "view" ? (
            <div className={styles.viewTitle}>{getValues("title")}</div>
          ) : (
            <Input
              type="text"
              register={register("title", {
                required: true,
              })}
              onBlur={(e) => setValue("title", e.target.value.trim())}
              disabled={operationType === "delete"}
            />
          )}
          {errors.title && (
            <span className={styles.error}>タイトルは必須です</span>
          )}

          {operationType === "view" ? (
            <div className={styles.viewLabel}>内容</div>
          ) : (
            <Label htmlFor="content" align="left" required>
              内容
            </Label>
          )}
          {operationType === "view" ? (
            <div className={styles.viewContent}>{getValues("content")}</div>
          ) : (
            <Textarea
              register={register("content", {
                required: true,
              })}
              maxLength={140}
              disabled={operationType === "delete"}
            />
          )}

          {errors.content && (
            <span className={styles.error}>内容は必須です</span>
          )}
        </div>

        <div className={operationType === "view" ? "" : styles.footer}>
          {
            {
              create: (
                <Button onClick={handleSubmit(onSubmit)}>新規投稿</Button>
              ),
              edit: <Button onClick={handleSubmit(onSubmit)}>編集</Button>,
              delete: (
                <Button buttonType="danger" onClick={handleDelete}>
                  削除
                </Button>
              ),
              view: <></>,
            }[operationType]
          }
        </div>
      </>
    </Dialog>
  );
}
