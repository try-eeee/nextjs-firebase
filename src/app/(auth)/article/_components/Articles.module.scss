.articlesHeader {
  display: flex;
  justify-items: end;
}

.editCell {
  display: flex;
  align-items: flex-start;

  svg {
    margin: 0.6rem;
  }

  @media (max-width: 768px) {
    justify-content: end;
  }
}
