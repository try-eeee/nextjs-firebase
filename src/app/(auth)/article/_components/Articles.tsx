"use client";

import {
  Table,
  TableCell,
  TableHeader,
  TableRow,
} from "@/components/table/Table";
// import { Button } from "@/components/button/Button";
import type { Article } from "../_utils/asyncFunctions";
import styles from "./Articles.module.scss";
// import { formatDate } from "@/utils/formatDate";
// import { useDialog } from "@/components/dialog/Dialog";
// import { useRef, useState } from "react";
// import { ArticleEditorModal, OperationType } from "./ArticleEditorModal";
// import { FaRegTrashAlt, FaEdit, FaRegEye } from "react-icons/fa";

interface ArticleProps {
  articles: Article[];
  uid: string;
}

export function Article(props: ArticleProps) {
  // const { articles, uid } = props;
  const { articles } = props;

  // const AricleRegisterModal = useRef<HTMLDialogElement>(null);
  // const { showModal } = useDialog(AricleRegisterModal);
  // const [operationType, setOperationType] = useState<OperationType>("create");
  // const [article, setArticle] = useState<Article | undefined>(undefined);

  if (!articles) {
    return <></>;
  }

  // const handleCreateArticle = () => {
  //   setOperationType("create");
  //   setArticle(undefined);
  //   showModal();
  // };

  // const handleDeleteArticle = (selectedArticle: Article) => {
  //   setOperationType("delete");
  //   setArticle(selectedArticle);
  //   showModal();
  // };

  // const handleEditArticle = (selectedArticle: Article) => {
  //   setOperationType("edit");
  //   setArticle(selectedArticle);
  //   showModal();
  // };

  // const handleViewArticle = (selectedArticle: Article) => {
  //   setOperationType("view");
  //   setArticle(selectedArticle);
  //   showModal();
  // };

  return (
    <>
      <div className={styles.articlesHeader}>
        {/* <Button buttonType="default" onClick={handleCreateArticle}>
          新規記事作成
        </Button> */}
      </div>
      <Table
        tableHeader={
          <TableRow>
            <TableHeader>記事ID</TableHeader>
            <TableHeader>更新日時</TableHeader>
            <TableHeader>タイトル</TableHeader>
            <TableHeader>閲覧 / 編集 / 削除</TableHeader>
          </TableRow>
        }
        tableRow={
          <>
            {articles.map((article, rowIndex) => (
              <TableRow key={rowIndex}>
                <TableCell headerLabel="記事ID" width={150}>
                  {article.articleId}
                </TableCell>
                <TableCell headerLabel="更新日時" width={150}>
                  {/* {formatDate(article.createdAt)} */}
                </TableCell>
                <TableCell headerLabel="タイトル">{article.title}</TableCell>
                <TableCell headerLabel="閲覧 / 編集 / 削除" width={180}>
                  {/* <div className={styles.editCell}>
                    <FaRegEye onClick={() => handleViewArticle(article)} />
                    {article.isSelf && (
                      <>
                        <FaEdit onClick={() => handleEditArticle(article)} />
                        <FaRegTrashAlt
                          onClick={() => handleDeleteArticle(article)}
                        />
                      </>
                    )}
                  </div> */}
                </TableCell>
              </TableRow>
            ))}
          </>
        }
      />
      {/* <ArticleEditorModal
        uid={uid}
        article={article}
        operationType={operationType}
        aricleRegisterModal={AricleRegisterModal}
      /> */}
    </>
  );
}
