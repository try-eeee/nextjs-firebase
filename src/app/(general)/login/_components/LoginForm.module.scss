.form {
  display: grid;
  grid-template-columns: 1fr;
  row-gap: 16px;
}

.title {
  font-size: 2rem;
  margin-bottom: 1.5rem;
}

.loginButton {
  margin-top: 14px;
  margin-bottom: 14px;
}

.error {
  color: red;
  font-size: 0.875rem;
  text-align: left;
}
