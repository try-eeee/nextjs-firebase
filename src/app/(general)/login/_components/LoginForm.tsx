"use client";

import { FormEvent, useState } from "react";
import Link from "next/link";
import { useRouter } from "next/navigation";
import { getAuth, signInWithEmailAndPassword } from "firebase/auth";
import { getFirebaseAuth } from "@/firebaseClient";
import { Input } from "@/components/input/Input";
import { Button } from "@/components/button/Button";
import { Label } from "@/components/label/Label";
import styles from "./LoginForm.module.scss";

/**
 * ログインフォーム
 */
export default function LoginForm() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [error, setError] = useState("");
  const router = useRouter();

  async function handleSubmit(event: FormEvent) {
    event.preventDefault();
    setError("");
    try {
      const { clientApp } = getFirebaseAuth();
      if (!clientApp) {
        throw new Error("Missing app!");
      }
      const credential = await signInWithEmailAndPassword(
        getAuth(clientApp),
        email,
        password
      );
      const idToken = await credential.user.getIdToken();
      await fetch("/api/login", {
        headers: { Authorization: `Bearer ${idToken}` },
      });
      router.push("/");
    } catch (e) {
      setError("ログイン情報が正しくありません。");
    }
  }

  return (
    <main>
      <h1 className={styles.title}>ログイン</h1>
      <form onSubmit={handleSubmit} className={styles.form} action="#">
        <div>
          <Label htmlFor="email" align="left">
            メールアドレス
          </Label>

          <Input
            type="email"
            name="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            id="email"
            className=""
            placeholder="name@company.com"
            required
          />
        </div>
        <div className={styles.mb5rem}>
          <Label htmlFor="password" align="left">
            パスワード
          </Label>
          <Input
            type="password"
            name="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            id="password"
            placeholder="••••••••"
            className=""
            required
          />
        </div>
        {error && (
          <div className={styles.error} role="alert">
            <span className="">{error}</span>
          </div>
        )}
        <Button type="submit" className={styles.loginButton}>
          ログイン
        </Button>
        <p>
          まだ会員登録がお済みでないですか?
          <br />
          <Link href="/register" className="">
            新規会員登録は こちら
          </Link>
        </p>
      </form>
    </main>
  );
}
