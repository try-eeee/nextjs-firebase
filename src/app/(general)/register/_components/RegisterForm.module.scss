.wrapper {
  padding: 1rem;
  margin: 1rem;
}

.form {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  gap: 1rem;
  max-width: 800px;
  margin: 0 auto;
}

.formItem {
  display: flex;
  flex-direction: column;
}

.fullWidth {
  grid-column: span 2;
}

.alignCenter {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
  text-align: center;
}

.alignLeft {
  text-align: left;
}

.title {
  font-size: 2rem;
  margin-bottom: 1.5rem;
}

.error {
  color: red;
  font-size: 0.875rem;
  text-align: left;
}
