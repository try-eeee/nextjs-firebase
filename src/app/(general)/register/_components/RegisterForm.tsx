"use client";

import { useState } from "react";
import Link from "next/link";
import {
  createUser,
  deleteUser,
  RegisterFormValues,
  updateUser,
} from "../../action";
import { useRouter } from "next/navigation";
import { FirebaseError } from "firebase-admin";
import { Label } from "@/components/label/Label";
import { Input } from "@/components/input/Input";
import { Button } from "@/components/button/Button";
import { useForm } from "react-hook-form";
import { zodResolver } from "@hookform/resolvers/zod";
import { clientRegisterSchema } from "../_utils/registerSchema";
import { RadioGroup } from "@/components/radioGroup/RadioGroup";
import { BirthdayInput } from "@/components/birthdayInput/BirthdayInput";
import { CheckBoxWithLabel } from "@/components/checkBoxWithLabel/CheckBoxWithLabel";
import styles from "./RegisterForm.module.scss";
import { FileUpload } from "@/components/fileUpload/FileUpload";
import { ref, uploadBytes, getDownloadURL } from "firebase/storage";
import { getFirebaseAuth } from "@/firebaseClient";

interface ExtendedError extends Error {
  code?: string;
}

const defaultValues = {
  lastName: "",
  firstName: "",
  lastNameKana: "",
  firstNameKana: "",
  gender: "",
  email: "",
  birthDate: "",
  password: "",
  confirmPassword: "",
  agreement: false,
  isUploadImage: false,
};

/**
 * 新規会員登録フォーム
 */
export function RegisterForm() {
  const router = useRouter();
  const [customErrors, setCustomErrors] = useState({
    matchedPassword: "",
    submit: "",
  });
  const [file, setFile] = useState<File | null>(null);

  const {
    register,
    handleSubmit,
    setValue,
    setError,
    clearErrors,
    getValues,
    formState: { errors },
  } = useForm<RegisterFormValues>({
    mode: "onSubmit",
    defaultValues,
    resolver: zodResolver(clientRegisterSchema),
  });

  const onSubmit = async (data: RegisterFormValues) => {
    setCustomErrors({
      matchedPassword: "",
      submit: "",
    });

    if (data.password !== data.confirmPassword) {
      setCustomErrors((prevErrors) => ({
        ...prevErrors,
        matchedPassword: "パスワードとパスワード(確認)が一致しません。",
      }));
      return;
    }

    let uid = "";
    try {
      uid = await createUser(data);
      const iconURL = await imageUpload(uid);

      if (!iconURL) {
        throw new Error("ダウンロードした画像取得に失敗しました");
      }

      await updateUser(uid, iconURL);

      router.push("/login");
    } catch (e) {
      const errorMessage = (e as FirebaseError).message;

      // 画像アップロードに失敗した場合、作成したユーザーを削除
      if (uid) {
        await deleteUser(uid);
      }

      setCustomErrors((prevErrors) => ({
        ...prevErrors,
        matchedPassword: errorMessage,
      }));
    }
  };

  const imageUpload = async (uid: string) => {
    try {
      if (!uid || !file) {
        throw new Error();
      }
      const { storage } = getFirebaseAuth();

      const fileName = `userIcon.${file.name.split(".").pop()}`;
      const storageRef = ref(storage, `userProfileIcon/${uid}/${fileName}`);

      // Firebase Storageに画像をアップロード
      const result = await uploadBytes(storageRef, file).then(
        async (snapshot) => {
          const downloadURL = await getDownloadURL(snapshot.ref);
          return downloadURL;
        }
      );

      return result;
    } catch (error) {
      if (!uid) {
        const error: ExtendedError = new Error(
          "アップロードの入力が不正です。"
        );
        error.code = "INVALID_UPLOAD_INPUT";
        throw error;
      }
    }
  };

  // アップロードされた際の処理
  const handleUploadUserIcon = () => {
    setValue("isUploadImage", true);
    clearErrors("isUploadImage");
  };

  // アップロードされた画像が削除された時の処理
  const handleRemoveUserIcon = () => {
    setValue("isUploadImage", false);
    setError("isUploadImage", {
      type: "manual",
      message: "ユーザーアイコンは必須です",
    });
  };

  return (
    <main className={styles.wrapper}>
      <h1 className={styles.title}>新規会員登録</h1>

      <div className={styles.form}>
        <div className={styles.formItem}>
          <Label htmlFor="lastName" align="left" required>
            姓
          </Label>

          <Input
            type="text"
            id="lastName"
            register={register("lastName", {
              required: true,
            })}
            onBlur={(e) => setValue("lastName", e.target.value.trim())}
          />
          {errors.lastName && (
            <span className={styles.error}>{errors.lastName.message}</span>
          )}
        </div>
        <div className={styles.formItem}>
          <Label htmlFor="firstName" align="left" required>
            名
          </Label>

          <Input
            type="text"
            id="firstName"
            register={register("firstName", { required: true })}
          />
          {errors.firstName && (
            <span className={styles.error}>{errors.firstName.message}</span>
          )}
        </div>
        <div className={styles.formItem}>
          <Label htmlFor="lastNameKana" align="left" required>
            セイ
          </Label>

          <Input
            type="text"
            id="lastNameKana"
            register={register("lastNameKana", { required: true })}
          />
          {errors.lastNameKana && (
            <span className={styles.error}>{errors.lastNameKana.message}</span>
          )}
        </div>
        <div className={styles.formItem}>
          <Label htmlFor="firstNameKana" align="left" required>
            メイ
          </Label>
          <Input
            type="text"
            id="firstNameKana"
            register={register("firstNameKana", { required: true })}
          />
          {errors.firstNameKana && (
            <span className={styles.error}>{errors.firstNameKana.message}</span>
          )}
        </div>
        <div className={`${styles.fullWidth}`}>
          <Label htmlFor="gender" align="left" required>
            姓別
          </Label>

          <RadioGroup
            options={[
              { value: "male", label: "男性" },
              { value: "female", label: "女性" },
              { value: "other", label: "その他" },
            ]}
            register={register("gender", { required: true })}
          />

          {errors.gender && (
            <div className={styles.error}>{errors.gender.message}</div>
          )}
        </div>
        <div className={styles.fullWidth}>
          <Label htmlFor="email" align="left" required>
            メールアドレス
          </Label>
          <Input
            type="email"
            id="email"
            register={register("email", { required: true })}
          />
          {errors.email && (
            <div className={styles.error}>{errors.email.message}</div>
          )}
        </div>
        <div className={`${styles.fullWidth} ${styles.alignLeft}`}>
          <Label htmlFor="birthDate" align="left" required>
            生年月日
          </Label>
          <BirthdayInput
            type="date"
            id="birthDate"
            register={register("birthDate", {
              required: true,
            })}
          />
          {errors.birthDate && (
            <div className={styles.error}>{errors.birthDate.message}</div>
          )}
        </div>
        <div className={styles.formItem}>
          <Label htmlFor="password" align="left" required>
            パスワード
          </Label>
          <Input
            type="password"
            id="password"
            register={register("password", { required: true })}
          />
          {errors.password && (
            <span className={styles.error}>{errors.password.message}</span>
          )}
        </div>
        <div className={styles.formItem}>
          <Label htmlFor="confirmPassword" align="left" required>
            パスワード（確認）
          </Label>
          <Input
            type="password"
            id="confirmPassword"
            register={register("confirmPassword", { required: true })}
          />
          {errors.confirmPassword && (
            <span className={styles.error}>
              {errors.confirmPassword.message}
            </span>
          )}
        </div>

        <div className={`${styles.fullWidth} ${styles.alignLeft}`}>
          <Label htmlFor="iconUpload" align="left" required>
            ユーザーアイコン
          </Label>
          <FileUpload
            setFile={setFile}
            handleUploadImage={handleUploadUserIcon}
            handleRemoved={handleRemoveUserIcon}
          />
          {errors.isUploadImage && (
            <div className={styles.error}>{errors.isUploadImage.message}</div>
          )}
        </div>

        <div className={`${styles.fullWidth} ${styles.alignCenter}`}>
          <CheckBoxWithLabel register={register("agreement")}>
            <>
              利用規約に同意する 利用規約は
              <a
                href="https://luna-matching.notion.site/a714620bbd8740d1ac98f2326fbd0bbc"
                target="_blank"
                rel="noopener noreferrer"
                className={styles.agreementLink}
              >
                こちら
              </a>
            </>
          </CheckBoxWithLabel>
          {errors.agreement && (
            <span className={styles.error}>{errors.agreement.message}</span>
          )}

          {customErrors.matchedPassword && (
            <span className={styles.error}>{customErrors.matchedPassword}</span>
          )}

          {customErrors.submit && (
            <span className={styles.error}>{customErrors.submit}</span>
          )}
          <Button onClick={handleSubmit(onSubmit)}>アカウント作成</Button>

          <p className={styles.fullWidth}>
            既にアカウントをお持ちですか?
            <Link href="/login" className={styles.loginLink}>
              ログインは <span>こちら</span>
            </Link>
          </p>
        </div>
      </div>
    </main>
  );
}
