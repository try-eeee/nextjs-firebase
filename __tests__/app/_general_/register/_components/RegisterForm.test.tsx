import React from "react";
import { render } from "@testing-library/react";
import { RegisterForm } from "@/app/(general)/register/_components/RegisterForm";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

jest.mock("@/app/(general)/actions", () => ({
  createUser: jest.fn(),
  deleteUser: jest.fn(),
  updateUser: jest.fn(),
}));

describe("RegisterForm UnitTest", () => {
  test("Snapshot Test Of RegisterForm", () => {
    const { asFragment } = render(<RegisterForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
