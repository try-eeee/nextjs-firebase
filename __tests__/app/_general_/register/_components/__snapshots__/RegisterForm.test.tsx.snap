// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm UnitTest Snapshot Test Of RegisterForm 1`] = `
<DocumentFragment>
  <main
    class="wrapper"
  >
    <h1
      class="title"
    >
      新規会員登録
    </h1>
    <div
      class="form"
    >
      <div
        class="formItem"
      >
        <label
          class="label left"
          for="lastName"
        >
          姓
          <span
            class="required"
          >
            必須
          </span>
        </label>
        <input
          class="input"
          id="lastName"
          name="lastName"
          type="text"
        />
      </div>
      <div
        class="formItem"
      >
        <label
          class="label left"
          for="firstName"
        >
          名
          <span
            class="required"
          >
            必須
          </span>
        </label>
        <input
          class="input"
          id="firstName"
          name="firstName"
          type="text"
        />
      </div>
      <div
        class="formItem"
      >
        <label
          class="label left"
          for="lastNameKana"
        >
          セイ
          <span
            class="required"
          >
            必須
          </span>
        </label>
        <input
          class="input"
          id="lastNameKana"
          name="lastNameKana"
          type="text"
        />
      </div>
      <div
        class="formItem"
      >
        <label
          class="label left"
          for="firstNameKana"
        >
          メイ
          <span
            class="required"
          >
            必須
          </span>
        </label>
        <input
          class="input"
          id="firstNameKana"
          name="firstNameKana"
          type="text"
        />
      </div>
      <div
        class="fullWidth"
      >
        <label
          class="label left"
          for="gender"
        >
          姓別
          <span
            class="required"
          >
            必須
          </span>
        </label>
        <div
          class="radioGroup"
        >
          <label
            class="radioLabel"
          >
            <input
              name="gender"
              type="radio"
              value="male"
            />
            <span
              class="radioButton"
            />
            男性
          </label>
          <label
            class="radioLabel"
          >
            <input
              name="gender"
              type="radio"
              value="female"
            />
            <span
              class="radioButton"
            />
            女性
          </label>
          <label
            class="radioLabel"
          >
            <input
              name="gender"
              type="radio"
              value="other"
            />
            <span
              class="radioButton"
            />
            その他
          </label>
        </div>
      </div>
      <div
        class="fullWidth"
      >
        <label
          class="label left"
          for="email"
        >
          メールアドレス
          <span
            class="required"
          >
            必須
          </span>
        </label>
        <input
          class="input"
          id="email"
          name="email"
          type="email"
        />
      </div>
      <div
        class="fullWidth alignLeft"
      >
        <label
          class="label left"
          for="birthDate"
        >
          生年月日
          <span
            class="required"
          >
            必須
          </span>
        </label>
        <input
          class="birthdayInput"
          id="birthDate"
          name="birthDate"
          type="date"
        />
      </div>
      <div
        class="formItem"
      >
        <label
          class="label left"
          for="password"
        >
          パスワード
          <span
            class="required"
          >
            必須
          </span>
        </label>
        <input
          class="input"
          id="password"
          name="password"
          type="password"
        />
      </div>
      <div
        class="formItem"
      >
        <label
          class="label left"
          for="confirmPassword"
        >
          パスワード（確認）
          <span
            class="required"
          >
            必須
          </span>
        </label>
        <input
          class="input"
          id="confirmPassword"
          name="confirmPassword"
          type="password"
        />
      </div>
      <div
        class="fullWidth alignLeft"
      >
        <label
          class="label left"
          for="iconUpload"
        >
          ユーザーアイコン
          <span
            class="required"
          >
            必須
          </span>
        </label>
        <label
          for="fileInput"
        >
          <button
            class="button default size-S"
          >
            画像をアップロード
          </button>
          <input
            accept="image/*"
            class="inputHidden"
            id="fileInput"
            type="file"
          />
        </label>
      </div>
      <div
        class="fullWidth alignCenter"
      >
        <label
          class="checkbox"
        >
          <input
            name="agreement"
            type="checkbox"
          />
          <span
            class="label"
          >
            利用規約に同意する 利用規約は
            <a
              class="agreementLink"
              href="https://luna-matching.notion.site/a714620bbd8740d1ac98f2326fbd0bbc"
              rel="noopener noreferrer"
              target="_blank"
            >
              こちら
            </a>
          </span>
        </label>
        <button
          class="button default size-S"
        >
          アカウント作成
        </button>
        <p
          class="fullWidth"
        >
          既にアカウントをお持ちですか?
          <a
            class="loginLink"
            href="/login"
          >
            ログインは 
            <span>
              こちら
            </span>
          </a>
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
